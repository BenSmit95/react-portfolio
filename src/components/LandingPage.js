import React from 'react';
import { Link } from 'react-router-dom';

class LandingPage extends React.Component {
    state = {
        toggleTexts: ['Creative', 'Curious', 'Innovative'],
        toggleTextIterator: 0,
        toggleTextId: 'undefined'
    }

    render() {
        // Which text is visible?
        const toggleTexts = this.state.toggleTexts.map((text, index) => (
            this.state.toggleTextIterator === index ? (
                <p key={index} className="landing__toggle-text">{text}</p>
            ) : (
                <p key={index} className="hidden landing__toggle-text">{text}</p>
            )
        ))

        return (
            <div className="landing">
                {toggleTexts}
                <div className="landing__message">
                    <h1>Hey there!</h1>
                    <h3>Welcome to my portfolio</h3>
                    <Link className ="button" to="/introduction">Proceed</Link>
                </div>
            </div>
        );
    }

    // after mounting, set a timer to toggle the text
    componentDidMount = () => {
        const toggleTextId = setInterval(() => {
            const toggleTextIterator = (this.state.toggleTextIterator + 1) % this.state.toggleTexts.length;
            this.setState(() => ({
                toggleTextIterator
            }));
        }, 3500)
        this.setState(() => ({
            toggleTextId
        }));
    }

    // Remove the toggle text when unmounting
    componentWillUnmount = () => {
        clearInterval(this.state.toggleTextId);
    }
};

export default LandingPage;