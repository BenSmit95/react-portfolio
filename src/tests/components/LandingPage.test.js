import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from '../../components/LandingPage';

test('Should render LandingPage component correctly', () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper).toMatchSnapshot();
});