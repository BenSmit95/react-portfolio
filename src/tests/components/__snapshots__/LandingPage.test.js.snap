// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LandingPage component correctly 1`] = `
<div
  className="landing"
>
  <p
    className="landing__toggle-text"
    key="0"
  >
    Creative
  </p>
  <p
    className="hidden landing__toggle-text"
    key="1"
  >
    Curious
  </p>
  <p
    className="hidden landing__toggle-text"
    key="2"
  >
    Innovative
  </p>
  <div
    className="landing__message"
  >
    <h1>
      Hey there!
    </h1>
    <h3>
      Welcome to my portfolio
    </h3>
    <Link
      className="button"
      replace={false}
      to="/introduction"
    >
      Proceed
    </Link>
  </div>
</div>
`;
